Automattic-Tracks-iOSTests/Tests/CrashLoggingTests.swift

```diff
@@ -9,176 +9,76 @@ class CrashLoggingTests: XCTestCase {
 
     override func setUp() {
         mockDataProvider.sentryDSN = validDSN
-        CrashLogging.isStarted = false  // Reset the Crash Logging system
     }
 
     override func tearDown() {
-        // A few of the test cases below use shouldSentEventCallback with an XCTestExpectation.
-        // Because they're using a singleton, we have to make sure to clear those to avoid
-        // dangling state and cross-pollination.
-        CrashLogging.sharedInstance.shouldSendEventCallback = nil
-
         mockDataProvider.reset()
     }
 
     func testInitializationWithInvalidDSN() {
-        let expectation = XCTestExpectation(description: "Intialization should fail with a DSN error")
-
-        mockDataProvider.sentryDSN = invalidDSN
-        mockDataProvider.didLogErrorCallback = { event in
-            if event.message == "Project ID path component of DSN is missing" {
-                expectation.fulfill()
-            }
+        do {
+            mockDataProvider.sentryDSN = "--invalid DSN--"
+            _ = try CrashLogging(dataProvider: mockDataProvider).start()
+            XCTFail("The call above should fail")
+        } catch let err {
+            XCTAssertEqual("Project ID path component of DSN is missing", err.localizedDescription)
         }
-
-        CrashLogging.start(withDataProvider: mockDataProvider)
-
-        wait(for: [expectation], timeout: 1)
     }
 
-    func testInitializationWithValidDSNFormat() {
-        let expectation = XCTestExpectation(description: "Intialization should pass")
-        expectation.isInverted = true
-
-        mockDataProvider.didLogErrorCallback = { event in
-            expectation.fulfill()
-        }
-
-        CrashLogging.start(withDataProvider: mockDataProvider)
-
-        wait(for: [expectation], timeout: 1)
+    func testInitializationWithValidDSNFormat() throws {
+        _ = try CrashLogging(dataProvider: mockDataProvider).start()
     }
 
-    func testThatUserDataIsBeingStoredForUseInCrashLogs() {
+    func testThatUserDataIsBeingPopulatedWhenSendingCrashLogs() throws {
         mockDataProvider.currentUser = testUser
-        CrashLogging.start(withDataProvider: mockDataProvider)
+        let crashLogging = try CrashLogging(dataProvider: mockDataProvider).start()
+        let event = crashLogging.beforeSend(event: Event(level: .debug))
 
-        XCTAssert(CrashLogging.sharedInstance.currentUser.email == testUser.email)
-        XCTAssert(CrashLogging.sharedInstance.cachedUser?.email == testUser.email)
+        XCTAssertEqual(testUser.email, event?.user?.email)
```

Comment:
Nitpick. It's unclear in the signature, but the convention I've always seen used with `XCTAssertEqual` it to have the actual value as the first element and the expect as the second.

That's what the Apple examples do, too:

<img width="812" alt="Screen Shot 2021-01-20 at 12 31 00 pm" src="https://user-images.githubusercontent.com/1218433/105114632-6021bb00-5b1b-11eb-96ee-2339c63f152b.png">

[source](https://developer.apple.com/documentation/xctest/2142776-xctassertequal)

![image](https://user-images.githubusercontent.com/1218433/105114672-762f7b80-5b1b-11eb-9421-ed2b59bb7b65.png)

[source](https://developer.apple.com/documentation/xctest/defining_test_cases_and_test_methods).

Is there a different convention in place in this codebase?

Reply:
No – that convention is correct!

Here, the `testUser` is the "constant" (ie – not the subject of the test) so the ordering is as expected.

Definitely something to nitpick – IMHO this is important for test readability so we should keep it consistent!!

Reply:
Is it though? @mokagio 's suggestion is that the convention is `XCTAssert(sut, expectedConstantValue)`, so the constant being as the second parameter. That's also always the convention I used I think.
So if the SUT is `event?.user?.email` and the `testUser` is the "constant" (= the expected value), then the order in your code is reversed, isn't it?